### PR TITLE
refactor: clang-tidy fixes to silence clangd warning

### PR DESCRIPTION
**Pull request by dundargoc**
- refactor: readability-uppercase-literal-suffix

- refactor: readability-named-parameter

- refactor: bugprone-suspicious-string-compare

- refactor: google-readability-casting

- refactor: readability-redundant-control-flow

- refactor: bugprone-too-small-loop-variable

- refactor: readability-non-const-parameter

- refactor: readability-avoid-const-params-in-decls

- refactor: google-readability-todo

- refactor: readability-inconsistent-declaration-parameter-name

- refactor: bugprone-suspicious-missing-comma

- refactor: remove noisy or slow warnings

### CODE REVIEW

**Comment by zeertzjq** [comment]
Most of these variable renames don't seem to make sense.

File: src/nvim/debugger.c [hunk 1]
```diff
@@ -50,7 +50,7 @@ struct debuggy {
 void do_debug(char *cmd)
 {
   int save_msg_scroll = msg_scroll;
-  int save_State = State;
+  int save_state = State;
```
Comment (bfredl) [inline src/nvim/debugger.c:53]: this only makes sense togheter with changing `State` to `state` (or something else, to make in more unique as a global var)

Comment (bfredl) [reply src/nvim/debugger.c:53]: we could skip this rule for now.

File: src/nvim/memline.c [hunk 2]
```diff
@@ -437,7 +437,7 @@ void ml_open_file(buf_T *buf)
   if (buf->b_spell) {
     char *fname = vim_tempname();
     if (fname != NULL) {
-      (void)mf_open_file(mfp, (char *)fname);           // consumes fname!
+      (void)mf_open_file(mfp, fname);           // consumes fname!
```
Comment (justinmk) [inline src/nvim/memline.c:440]: nice that we're getting warnings for this

File: src/nvim/tag.c [hunk 3]
```diff
@@ -642,9 +642,7 @@ void do_tag(char *tag, int type, int count, int forceit, int verbose)
     curwin->w_tagstackidx = tagstackidx;
   }
   postponed_split = 0;          // don't split next time
-  g_do_tagpreview = 0;          // don't do tag preview next time
-
-  return;
+  g_do_tagpreview = 0;
```
Comment (justinmk) [inline src/nvim/tag.c:645]: comment was deleted?

**Review by justinmk — APPROVED** [review]

File: src/nvim/getchar.c [hunk 4]
```diff
@@ -509,7 +509,7 @@ void AppendToRedobuffLit(const char *str, int len)
       s--;
     }
     if (s > start) {
-      add_buff(&redobuff, start, (long)(s - start));
+      add_buff(&redobuff, start, (long)(s - start));  // NOLINT(google-readability-casting)
```
Comment (zeertzjq) [inline src/nvim/getchar.c:512]: It seems that this cast can indeed be removed, as `add_buff()` accepts a `ptrdiff_t`.

File: src/nvim/eval/funcs.c [hunk 5]
```diff
@@ -6022,17 +6022,17 @@ static void f_readfile(typval_T *argvars, typval_T *rettv, EvalFuncData fptr)
...
-          long growmin  = (long)((p - start) * 2 + prevlen);
+          long growmin  = (long)((p - start) * 2 + prevlen);  // NOLINT(google-readability-casting)
           prevsize = grow50pc > growmin ? grow50pc : growmin;
         }
         prev = xrealloc(prev, (size_t)prevsize);
       }
       // Add the line part to end of "prev".
       memmove(prev + prevlen, start, (size_t)(p - start));
-      prevlen += (long)(p - start);
+      prevlen += (long)(p - start);  // NOLINT(google-readability-casting)
```
Comment (zeertzjq) [inline src/nvim/eval/funcs.c:6035]: It's also possible to change `prevlen` `prevsize` `grow50pc` `growmin` all to `ptrdiff_t` and remove this cast.

File: src/nvim/statusline.c [hunk 6]
```diff
@@ -929,7 +929,7 @@ int build_stl_str_hl(win_T *wp, char *out, size_t outlen, char *fmt, int use_san
           group_len = (min_group_width - group_len) * utf_char2len(fillchar);
           memmove(t + group_len, t, (size_t)(out_p - t));
           if (out_p + group_len >= (out_end_p + 1)) {
-            group_len = (long)(out_end_p - out_p);
+            group_len = (long)(out_end_p - out_p);  // NOLINT(google-readability-casting)
```
Comment (zeertzjq) [inline src/nvim/statusline.c:932]: It's also possible to change `group_len` and `min_group_width` to `ptrdiff_t` and remove the cast here.

**Review by zeertzjq — APPROVED** [review]

File: src/nvim/version.c [hunk 7]
```diff
@@ -2218,7 +2218,7 @@ void intro_message(int colon)
   int sponsor;
   char *p;
   static char *(lines[]) = {
-    N_(NVIM_VERSION_LONG),
+    N_(NVIM_VERSION_LONG),  // NOLINT(bugprone-suspicious-missing-comma)
```
Comment (justinmk) [inline src/nvim/version.c:2221]: shouldn't this be done at the definition, i.e. here: https://github.com/neovim/neovim/blob/ba887da97708545f1174e387b4807119ad5a40f6/src/nvim/version.c#L37

Comment (dundargoc) [reply src/nvim/version.c:2221]: Doesn't matter, both works. I don't think using `//` is necessary, but it should work as well.

**Review by justinmk — APPROVED** [review]

**Pull request merged**